src/platform/plugins/shared/discover/public/application/main/components/layout/cascaded_documents/blocks/cascade_leaf_component.tsx

```diff
@@ -0,0 +1,338 @@
+  const virtualizerScrollElementRef = useRef<Element | null>(getScrollElement());
+
+  const scrollElementGetter = useCallback(() => virtualizerScrollElementRef.current, []);
+
+  useEffect(() => {
+    const defaultScrollElement = getScrollElement();
+
+    if (
+      isFullScreenMode &&
+      customGridBodyScrollContainerRef.current &&
+      !virtualizerScrollElementRef.current?.isSameNode(customGridBodyScrollContainerRef.current)
+    ) {
+      // assign the custom grid body element as scrollable element in full screen mode
+      // customCascadeGridScrollRef.current = $gridBody;
+      virtualizerScrollElementRef.current = customGridBodyScrollContainerRef.current;
+    } else if (
+      !isFullScreenMode &&
+      !virtualizerScrollElementRef.current?.isSameNode(defaultScrollElement)
+    ) {
+      // when exiting full screen mode we ensure the scroll element is reset to the default scroll container
+      virtualizerScrollElementRef.current = defaultScrollElement;
+    }
+  }, [getScrollElement, isFullScreenMode]);
+
+  const virtualizer = useVirtualizer({
+    count: visibleRows.length,
+    estimateSize: () => 50,
+    overscan: 15,
+    initialOffset,
+    scrollMargin,
+    getScrollElement: scrollElementGetter,
+    useAnimationFrameWithResizeObserver: true,
+  });
+
+  const items = virtualizer.getVirtualItems();
+
+  return (
+    <div
+      data-test-subj="discoverCascadeCustomDataGridBody"
+      role="rowgroup"
+      css={customCascadeGridBodyStyle.wrapper}
+    >
+      <div css={customCascadeGridBodyStyle.headerRow}>{headerRow}</div>
+      <div
+        ref={customGridBodyScrollContainerRef}
+        css={customCascadeGridBodyStyle.virtualizerContainer}
+      >
+        <div
+          style={{
+            height: virtualizer.getTotalSize(),
+          }}
+        >
+          <div
+            css={customCascadeGridBodyStyle.virtualizerInner}
+            style={{
+              transform: `translateY(${
+                items[0]?.start - (virtualizer.options.scrollMargin ?? 0)
+              }px)`,
+            }}
+          >
+            {items.map((virtualRow) => (
+              <div
+                key={virtualRow.key}
+                data-index={virtualRow.index}
+                ref={virtualizer.measureElement}
+                className="euiDataGridRow"
+                css={customCascadeGridBodyStyle.displayFlex}
+              >
+                {visibleColumns.map((column, colIndex) => {
+                  return (
+                    <Cell
+                      key={`${virtualRow.index}-${colIndex}`}
+                      colIndex={colIndex}
+                      rowIndex={virtualRow.index}
+                      visibleRowIndex={virtualRow.index}
+                      columnId={column.id}
+                    />
+                  );
+                })}
+              </div>
+            ))}
+          </div>
+        </div>
+      </div>
+      <React.Fragment>{footerRow}</React.Fragment>
```

Comment:
```suggestion
      {footerRow}
```
Nit: Is the fragment needed here?